src/testing/puppeteer/puppeteer-declarations.ts

```diff
@@ -1,14 +1,45 @@
 import type { EventInitDict, EventSpy, ScreenshotDiff, ScreenshotOptions } from '@stencil/core/internal';
 import type {
   ClickOptions,
-  NavigationOptions,
+  HTTPResponse as PuppeteerHTTPResponse,
   Page,
-  PageCloseOptions,
   ScreenshotOptions as PuppeteerScreenshotOptions,
-  Response,
 } from 'puppeteer';
 
-export interface NewE2EPageOptions extends NavigationOptions {
+/**
+ * This type helps with declaration merging as a part of Stencil's migration from Puppeteer v5.4.3 to v10.0.0. In
+ * v5.4.3, `HttpResponse` was an interface whereas v10.0.0 declares it as a class. It is redeclared here to help teams
+ * migrate to a newer version of Stencil minor version without requiring a Puppeteer upgrade. This type should be
+ * removed as a part of the Stencil 3.0 release.
+ */
+export type HTTPResponse = PuppeteerHTTPResponse;
+
+/**
+ * These types help with declaration merging as a part of Stencil's migration from Puppeteer v5.4.3 to v10.0.0. In
+ * v10.0.0, `WaitForOptions` is a renamed version of `NavigationOptions` from v5.4.3, who has had its type hierarchy
+ * flattened.
+ *
+ * See {@link https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8290e943f6b398acf39ee1b2e486824144e15bc8/types/puppeteer/index.d.ts#L605-L622}
+ * for the v5.4.3 types.
+ *
+ * These types are redeclared here to help teams migrate to a newer version of Stencil minor version without requiring a
+ * Puppeteer upgrade. This type should be removed as a part of the Stencil 3.0 release.
+ */
+export type PuppeteerLifeCycleEvent = 'load' | 'domcontentloaded' | 'networkidle0' | 'networkidle2';
+export interface WaitForOptions {
+  timeout?: number;
+  waitUntil?: PuppeteerLifeCycleEvent | PuppeteerLifeCycleEvent[];
+}
```

Comment:
You may be able to declare these on the puppeteer module:
```suggestion
declare module "puppeteer" {
    type LifeCycleEvent = 'load' | 'domcontentloaded' | 'networkidle0' | 'networkidle2';
    interface WaitForOptions {
        timeout?: number;
        waitUntil?: LifeCycleEvent | LifeCycleEvent[];
    }
}
```

Reply:
After a discussion offline, I've reverted this commit. We intend on this change to be a part of the next major version of Stencil. I've updated the target branch from `master` to `v3.0.0-dev`